File: MetaplaySDK/Plugins/AndroidCustomTabs/Package/Plugins/Android/MetaplayAndroidCustomTabs.cs
// This file is part of Metaplay SDK which is released under the Metaplay SDK License.

using Metaplay.Core;
using System;
using UnityEngine;

namespace Metaplay.Unity.Android
{
    public static class MetaplayAndroidCustomTabs
    {
        public static void LaunchCustomTabs(string url)
        {
            if (!MetaplayCore.Options.FeatureFlags.EnableCompanyAccountClientIntegrations)
                throw new InvalidOperationException("FeatureFlags.EnableCompanyAccountClientIntegrations must be enabled to use MetaplayAndroidCustomTabs.");

            using (AndroidJavaClass clazz = new AndroidJavaClass("io.metaplay.android.customtabs.MetaplayAndroidCustomTabs"))
            {
                clazz.CallStatic("LaunchCustomTabs", url);
            }
        }
    }
}

File: MetaplaySDK/Plugins/AndroidCustomTabs/Lib/metaplaycustomtabs/src/main/java/io/metaplay/android/customtabs/MetaplayAndroidCustomTabs.java
// This file is part of Metaplay SDK which is released under the Metaplay SDK License.
package io.metaplay.android.customtabs;

import android.net.Uri;
import androidx.browser.customtabs.*;
import com.unity3d.player.*;

public final class MetaplayAndroidCustomTabs
{
    public static void LaunchCustomTabs(String url)
    {
        CustomTabsIntent intent =
            new CustomTabsIntent.Builder()
            .setShowTitle(false)
            .setStartAnimations(UnityPlayer.currentActivity.getApplicationContext(), R.anim.customtabs_tabs_enter, R.anim.customtabs_app_exit)
            .setExitAnimations(UnityPlayer.currentActivity.getApplicationContext(), R.anim.customtabs_app_enter, R.anim.customtabs_tabs_exit)
            .build();
        intent.launchUrl(UnityPlayer.currentActivity, Uri.parse(url));
    }
}
